Dockerfile

```diff
@@ -6,54 +6,31 @@
 # docker run --init -it forest
 # ```
 # 
-# Build and manually push to Github Container Registry (see https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry)
-# ```
-# docker build -t ghcr.io/chainsafe/forest:latest .
-# docker push ghcr.io/chainsafe/forest:latest
-# ```
-
-##
-# Build stage
-# Use github action runner cached images to avoid being rate limited
-# https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2004-Readme.md#cached-docker-images
-## 
-
-# Cross-compilation helpers
-# https://github.com/tonistiigi/xx
-FROM --platform=$BUILDPLATFORM tonistiigi/xx:1.5.0 AS xx
 
-FROM --platform=$BUILDPLATFORM ubuntu:22.04 AS build-env
+FROM ubuntu:22.04 AS build-env
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
 # install dependencies
 RUN apt-get update && \
     apt-get install --no-install-recommends -y build-essential clang curl git ca-certificates
 RUN update-ca-certificates
 
+# install Rust
 RUN curl https://sh.rustup.rs -sSf | sh -s -- -y --no-modify-path --profile minimal
 ENV PATH="/root/.cargo/bin:${PATH}"
 
-# Copy the cross-compilation scripts 
-COPY --from=xx / /
-
-# export TARGETPLATFORM
-ARG TARGETPLATFORM
-
-# Install those packages for the target architecture
-RUN xx-apt-get update && \
-    xx-apt-get install -y libc6-dev g++
+# install Go.
+RUN curl -sSL https://raw.githubusercontent.com/voidint/g/master/install.sh | bash
```

Comment:
1. `master` is not deterministic and can be modified with whatever malicious code.
2. I don't really know this. Can we use a more official source?

Reply:
Pinned to v1.7.0

Reply:
What about the second point?

Reply:
> What about the second point?

@LesnyRumcajs Go does not offer an official version manager like `rustup`. I can switch to using go docker image to avoid 3rd party version managers